### PR TITLE
Correct mistakes

### PR DESCRIPTION
**Pull request by kasyanyukd1995**
good job

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -21,97 +22,78 @@ class MyApp extends StatelessWidget {
         // Notice that the counter didn't reset back to zero; the application
         // is not restarted.
         primarySwatch: Colors.blue,
+
         // This makes the visual density adapt to the platform that you run
         // the app on. For desktop platforms, the controls will be smaller and
```
Comment (Platton96) [inline lib/main.dart]: remove comments

File: lib/main.dart [hunk 2]
```diff
@@ -21,97 +22,78 @@ class MyApp extends StatelessWidget {
...
-      // This call to setState tells the Flutter framework that something has
-      // changed in this State, which causes it to rerun the build method below
-      // so that the display can reflect the updated values. If we changed
-      // _counter without calling setState(), then the build method would not be
-      // called again, and so nothing would appear to happen.
-      _counter++;
-    });
+class _MyAppState extends State<MyApp> {
+  Future<String> futureData;
+  //var a = await fetchRequest();
```
Comment (Platton96) [inline lib/main.dart]: remove

File: lib/main.dart [hunk 3]
```diff
@@ -21,97 +22,78 @@ class MyApp extends StatelessWidget {
...
-      home: MyHomePage(title: 'Flutter Demo Home Page'),
+      home: HomePage(),
     );
   }
 }
 
-class MyHomePage extends StatefulWidget {
-  MyHomePage({Key key, this.title}) : super(key: key);
+/*
+
```
Comment (Platton96) [inline lib/main.dart]: remove useless emptyline

File: lib/main.dart [hunk 4]
```diff
@@ -21,97 +22,78 @@ class MyApp extends StatelessWidget {
...
+
+Future<String> fetchRequest() async {
```
Comment (Platton96) [inline lib/main.dart]: rename to fetchMovies

File: lib/models/move_model.dart [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'move_model.g.dart';
+
+@JsonSerializable()
+class MoveModel {
```
Comment (Platton96) [inline lib/models/move_model.dart]: rename to MovieModel

File: lib/pages/actor_info_page.dart [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+              ),
+              Text(
+                actorModel.asCharacter,
+                style: TextStyle(
+                  color: Colors.white54,
+                ),
+              ),
+            ],
+          ),
+        )),
```
Comment (Platton96) [inline lib/pages/actor_info_page.dart]: add comma

File: lib/pages/actor_info_page.dart [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        //backgroundColor: Colors.white,
```
Comment (Platton96) [inline lib/pages/actor_info_page.dart]: remove comment

File: lib/pages/home_page.dart [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    Widget child;
+    switch (_index) {
+      case 0:
+        child = MostPopularMoviesPage();
+        break;
+      case 1:
+        child = MoviesPage();
+        break;
+      /*case 2:
+        child = Top250Page();
```
Comment (Platton96) [inline lib/pages/home_page.dart]: remove comment

File: lib/pages/home_page.dart [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+      backgroundColor: Colors.black,
+      appBar: AppBarWidget(),
+      body: SizedBox.expand(child: child),
+      bottomNavigationBar: BottomNavigationBar(
+        backgroundColor: Colors.black,
+        iconSize: 23,
+        selectedItemColor: Colors.amber,
+        onTap: (newIndex) => setState(() => _index = newIndex),
+        currentIndex: _index,
+
```
Comment (Platton96) [inline lib/pages/home_page.dart]: remove comment and space

File: lib/pages/home_page.dart [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        //fixedColor: Colors.black,
+        items: [
+          BottomNavigationBarItem(
+            icon: Icon(
+              Icons.home,
+              color: Colors.white,
```
Comment (Platton96) [inline lib/pages/home_page.dart]: create widget for it with  param iconData, string

File: lib/models/similar_model.dart [hunk 11]
```diff
@@ -0,0 +1,37 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'similar_model.g.dart';
+
+@JsonSerializable()
+class SimilarModel {
```
Comment (Platton96) [inline lib/models/similar_model.dart]: rename to similarMovieModel

File: lib/models/title_data_model.dart [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+  final String plot;
+  final String directors;
+  final String writers;
+  final String stars;
+  final List<ActorModel> actorList;
+  final String genres;
+  final String countries;
+  final String imDbRating;
+
+  final TrailerDataModel trailer;
```
Comment (Platton96) [inline lib/models/title_data_model.dart]: rename TrailerModel

File: lib/models/youtube_trailer_data_model.dart [hunk 13]
```diff
@@ -0,0 +1,20 @@
+import 'package:json_annotation/json_annotation.dart';
+part 'youtube_trailer_data_model.g.dart';
+
+@JsonSerializable()
+class YouTubeTrailerDataModel {
```
Comment (Platton96) [inline lib/models/youtube_trailer_data_model.dart]: YouTubeTrailerDataModel to YouTubeTrailerModel

File: lib/pages/home_page.dart [hunk 14]
```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+import 'package:kinopoisk/pages/index.dart';
+import 'package:kinopoisk/widgets/index.dart';
+
+class HomePage extends StatefulWidget {
+  //HomePage({Key key}) : super(key: key);
```
Comment (Platton96) [inline lib/pages/home_page.dart]: what is it?

File: lib/pages/home_page.dart [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+        selectedItemColor: Colors.amber,
+        onTap: (newIndex) => setState(() => _index = newIndex),
+        currentIndex: _index,
+        items: [
+          BottomNavigationBarItem(
+            icon: Icon(
+              Icons.home,
+              color: Colors.white,
+            ),
+            // ignore: deprecated_member_use
```
Comment (Platton96) [inline lib/pages/home_page.dart:46]: remove comment

File: lib/pages/most_popular_movies_page.dart [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+            constraints: BoxConstraints.expand(),
+            child: CarouselSlider.builder(
+              itemCount: snapshot.data.items.length,
+              options: CarouselOptions(
+                autoPlay: true,
+                enlargeCenterPage: true,
+                aspectRatio: 2.0,
+                height: double.infinity,
+              ),
+              itemBuilder: (ctx, index) {
```
Comment (Platton96) [inline lib/pages/most_popular_movies_page.dart]: write context or _ instead of ctx

File: lib/pages/movies_page.dart [hunk 17]
```diff
@@ -0,0 +1,199 @@
...
+            'Most Popular Movies Now',
+            style: TextStyle(
+              color: Colors.white,
+              fontSize: 26,
+              fontStyle: FontStyle.normal,
+              fontWeight: FontWeight.w400,
+            ),
+          ),
+          SizedBox(
+            height: 20.0,
```
Comment (Platton96) [inline lib/pages/movies_page.dart]: don't use comma with SizedBox()

File: lib/pages/movies_page.dart [hunk 18]
```diff
@@ -0,0 +1,199 @@
...
+                              movieItem: movieItem,
+                              onTapMovieFunction: (movieobj) => Navigator.push(
+                                  context,
+                                  MaterialPageRoute(
+                                      builder: (BuildContext context) =>
+                                          MoveInfoPage(
+                                            titleId: movieItem.id,
+                                            rating: movieItem.imDbRating != ''
+                                                ? movieItem.imDbRating
+                                                : null,
```
Comment (Platton96) [inline lib/pages/movies_page.dart:59]: add comma

File: lib/pages/movies_page.dart (hunk 17 above)
Comment (Platton96) [reply lib/pages/movies_page.dart]: and please add prefer_const_constructors in alnalysis_option

File: lib/pages/movies_page.dart [hunk 19]
```diff
@@ -0,0 +1,199 @@
...
+class MoviesPage extends StatefulWidget {
+  MoviesPage({Key key}) : super(key: key);
+
+  @override
+  _MoviesPageState createState() => _MoviesPageState();
+}
+
+class _MoviesPageState extends State<MoviesPage> {
+  @override
+  Widget build(BuildContext context) {
```
Comment (Platton96) [inline lib/pages/movies_page.dart:17]: the too big method, please separate small private methods

File: lib/pages/movies_page.dart [hunk 20]
```diff
@@ -0,0 +1,199 @@
...
+                                          MoveInfoPage(
+                                            titleId: movieItem.id,
+                                            rating: movieItem.imDbRating != ''
+                                                ? movieItem.imDbRating
+                                                : null,
+                                          ))),
+                            );
+                          },
+                          itemCount: snp.data.items.length > 30
+                              ? 30
```
Comment (Platton96) [inline lib/pages/movies_page.dart]: add constant for 30

File: lib/pages/movies_page.dart [hunk 21]
```diff
@@ -0,0 +1,199 @@
...
+                              onTapMovieFunction: (movieobj) => Navigator.push(
+                                  context,
+                                  MaterialPageRoute(
+                                      builder: (BuildContext context) =>
+                                          MoveInfoPage(
+                                            titleId: movieItem.id,
+                                            rating: movieItem.imDbRating != ''
+                                                ? movieItem.imDbRating
+                                                : null,
+                                          ))),
```
Comment (Platton96) [inline lib/pages/movies_page.dart:118]: add comma

File: lib/pages/movies_page.dart [hunk 22]
```diff
@@ -0,0 +1,199 @@
...
+                            );
+                          },
+                          itemCount: snp.data.items.length > 30
+                              ? 30
```
Comment (Platton96) [inline lib/pages/movies_page.dart]: add constants

File: lib/pages/movies_page.dart [hunk 23]
```diff
@@ -0,0 +1,199 @@
...
+                              onTapMovieFunction: (movieobj) => Navigator.push(
+                                  context,
+                                  MaterialPageRoute(
+                                      builder: (BuildContext context) =>
+                                          MoveInfoPage(
+                                            titleId: movieItem.id,
+                                            rating: movieItem.imDbRating != ''
+                                                ? movieItem.imDbRating
+                                                : null,
+                                          ))),
```
Comment (Platton96) [inline lib/pages/movies_page.dart]: add comma

File: lib/pages/search_page.dart [hunk 24]
```diff
@@ -0,0 +1,67 @@
...
+import 'package:http/http.dart' as http;
+import 'package:flutter/material.dart';
+import 'package:kinopoisk/data/apikey.dart';
+import 'package:kinopoisk/models/index.dart';
+import 'package:kinopoisk/pages/index.dart';
+import 'package:kinopoisk/widgets/index.dart';
+
+class SearchPage extends StatelessWidget {
+  Future<List<ResultSearchListModel>> search(String search) async {
+    await Future.delayed(Duration(seconds: 3));
```
Comment (Platton96) [inline lib/pages/search_page.dart]: is a bad approach to use  Future.delayed

File: lib/pages/search_page.dart [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+            ),
+            icon: Icon(
+              Icons.search,
+              color: Colors.white,
+              size: 30,
+            ),
+            shrinkWrap: true,
+            onSearch: search,
+            onItemFound: (ResultSearchListModel item, int index) {
+              return ItemForSearchWidget(
```
Comment (Platton96) [inline lib/pages/search_page.dart]: rename widget. for example SearchWidget

File: lib/widgets/actor_widget.dart [hunk 26]
```diff
@@ -0,0 +1,62 @@
...
+                              Icon(Icons.error),
+                          fit: BoxFit.fill,
+                          fadeInCurve: Curves.easeIn,
+                          fadeInDuration: Duration(seconds: 2),
+                          fadeOutCurve: Curves.easeOut,
+                          fadeOutDuration: Duration(seconds: 2),
+                        ),
+                ),
+                SizedBox(
+                  height: 6,
```
Comment (Platton96) [inline lib/widgets/actor_widget.dart]: don't use comma

File: lib/widgets/director_and_writerrs_info_widget.dart [hunk 27]
```diff
@@ -0,0 +1,133 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:kinopoisk/models/title_data_model.dart';
+
+class DirectorAndWritersinfoWidget extends StatelessWidget {
```
Comment (Platton96) [inline lib/widgets/director_and_writerrs_info_widget.dart]: rename to DirectorAndWritersInfoWidget

File: lib/widgets/director_and_writerrs_info_widget.dart [hunk 28]
```diff
@@ -0,0 +1,133 @@
...
+  final TitleDataModel titleItem;
+
+  const DirectorAndWritersinfoWidget({
+    this.titleItem,
+  });
+
+  Widget build(BuildContext context) {
+    if (titleItem.directors != '') {
```
Comment (Platton96) [inline lib/widgets/director_and_writerrs_info_widget.dart]: i dont like such 'ifs'

File: lib/widgets/director_and_writerrs_info_widget.dart (hunk 27 above)
Comment (Platton96) [reply lib/widgets/director_and_writerrs_info_widget.dart]: can you discuss solution with me

File: lib/widgets/director_and_writerrs_info_widget.dart [hunk 29]
```diff
@@ -0,0 +1,133 @@
...
+      if (titleItem.writers != '') {
+        return Column(
+          children: [
+            Align(
```
Comment (Platton96) [inline lib/widgets/director_and_writerrs_info_widget.dart]: you can not use all align

Comment (Platton96) [reply lib/widgets/director_and_writerrs_info_widget.dart]: column has property

File: lib/widgets/director_and_writerrs_info_widget.dart (hunk 27 above)
Comment (Platton96) [reply lib/widgets/director_and_writerrs_info_widget.dart]: this widget you should refactor

File: lib/widgets/image_widget.dart [hunk 30]
```diff
@@ -0,0 +1,66 @@
...
+                    fit: BoxFit.fill,
+                    fadeInCurve: Curves.easeIn,
+                    fadeInDuration: Duration(seconds: 2),
+                    fadeOutCurve: Curves.easeOut,
+                    fadeOutDuration: Duration(seconds: 2),
+                  ),
+          ),
+        ),
+        SizedBox(
+          width: 8,
```
Comment (Platton96) [inline lib/widgets/image_widget.dart]: remove comma

File: lib/pages/movies_page.dart (hunk 18 above)
Comment (kasyanyukd1995) [reply lib/pages/movies_page.dart:59]: what is it comma?

**Pull request merged**